File: src/main/java/ru/kosdev/signfinder/transform/RotateTransformer.java
package ru.kosdev.signfinder.transform;

import org.bytedeco.javacpp.opencv_core;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import static org.bytedeco.javacpp.opencv_imgproc.getRotationMatrix2D;
import static org.bytedeco.javacpp.opencv_imgproc.warpAffine;

/**
 * Вращение на заданный угол
 */
public class RotateTransformer implements Transformer {

    private final Logger log = LoggerFactory.getLogger(this.getClass());

    private double angle;

    public RotateTransformer(double angle) {
        log.info("Initializing rotate transformer with " + angle + " angle");
        this.angle = angle;
    }

    @Override
    public void accept(opencv_core.Mat mat) {
        //определяем центр (ось вращения)
        opencv_core.Point2f center = new opencv_core.Point2f(mat.cols() / 2, mat.rows() / 2);
        //матрица вращения
        opencv_core.Mat rotationMatrix = getRotationMatrix2D(center, angle, 1);
        warpAffine(mat, mat, rotationMatrix, mat.size());
    }
}

File: src/main/kotlin/ru/kosdev/signfinder/transform/RotateTransformer.kt
package ru.kosdev.signfinder.transform

import org.bytedeco.javacpp.opencv_core
import org.slf4j.Logger
import org.slf4j.LoggerFactory

import org.bytedeco.javacpp.opencv_imgproc.getRotationMatrix2D
import org.bytedeco.javacpp.opencv_imgproc.warpAffine
/**
 * Created by Константин on 27.09.2015.
 */
public class RotateTransformer(private val angle: Double) : Transformer {

    private val log = LoggerFactory.getLogger(this.javaClass)

    init {
        log.info("Initializing rotate transformer with $angle angle")
    }

    override fun accept(mat: opencv_core.Mat) {
        //определяем центр (ось вращения)
        val center = opencv_core.Point2f((mat.cols() / 2).toFloat(), (mat.rows() / 2).toFloat())
        //матрица вращения
        val rotationMatrix = getRotationMatrix2D(center, angle, 1.0)
        warpAffine(mat, mat, rotationMatrix, mat.size())
    }
}